### PR TITLE
Remove the Kafka.spec.topicOperator logic from the code

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

This PR implements the proposal #3091 and removes the logic behind the `Kafka.spec.topicOperator`. It leaves only the API to not break the compatibility. When `Kafka.spec.topicOperator` is specified, it adds a new `Warning` condition to status. The fact that it is removed is also already logged in the logs.

In addition, this adds a new simple test for the new warning condition and removes the topic operator from different tests.

No other references seem to exist in docs - at least I havn't found any.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 
 # CHANGELOG
 
+## 0.19.0
+
+* Remove the code behind `Kafka.spec.topicOperator` which did not work anymore
```
Comment (samuel-hawker) [inline CHANGELOG.md:6]: Maybe just say 
```
Remove deprecated `Kafka.spec.topicOperator` classes and deployment logic 
```
?

Comment (scholzj) [reply CHANGELOG.md]: Fixed.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorWithDeprecatedTopicOperatorTest.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(VertxExtension.class)
+public class KafkaAssemblyOperatorWithDeprecatedTopicOperatorTest {
```
Comment (samuel-hawker) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorWithDeprecatedTopicOperatorTest.java:43]: I wonder, if, going forward renaming this to something like `KafkaAssemblyOperatorUnsupportedFieldsTest` might be more extensible and then we just add to this file

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorWithDeprecatedTopicOperatorTest.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        vertx = Vertx.vertx();
+    }
+
+    @AfterAll
+    public static void after() {
+        vertx.close();
+    }
+
+    @Test
+    public void testNoTopicOperatorWarnings(VertxTestContext context) throws ParseException {
```
Comment (samuel-hawker) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorWithDeprecatedTopicOperatorTest.java:64]: Might be worth a doc comment something like
```
/**
 * spec.topicOperator is now unsupported, to ensure the removal does not cause unwanted warnings for users who  
 * do not use the topicOperator. This test checks this
 */
@Test
    public void testNoTopicOperatorWarnings(VertxTestContext context) throws ParseException {
```

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorWithDeprecatedTopicOperatorTest.java [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+                config);
+
+        Checkpoint async = context.checkpoint();
+        kao.createOrUpdate(new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, namespace, clusterName), kafka)
+                .onComplete(context.succeeding(v -> context.verify(() -> {
+                    assertThat(kafkaCaptor.getValue(), is(notNullValue()));
+                    assertThat(kafkaCaptor.getValue().getStatus(), is(notNullValue()));
+                    KafkaStatus status = kafkaCaptor.getValue().getStatus();
+
+                    assertThat(status.getConditions().size(), is(1));
```
Comment (samuel-hawker) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorWithDeprecatedTopicOperatorTest.java:112]: Checking conditions size may be a little flaky going forward if we add more default conditions but there is probably no way around this.

**Comment by samuel-hawker** [comment]
All looks good to me, left a few comments but are by no means a blocker to merge

File: design/remove-deprecated-topic-operator-from-kafka-crd.md [hunk 5]
```diff
@@ -1,8 +1,8 @@
 # Remove deprecated Topic Operator deployment in Kafka CRD
 
-| State    |
-|----------|
-| Approved |
+| State       |
+|-------------|
+| Implemented |
```
Comment (ppatierno) [inline design/remove-deprecated-topic-operator-from-kafka-crd.md:5]: did we decide to have the state in the design doc at the end?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorWithDeprecatedTopicOperatorTest.java (hunk 4 above)
Comment (scholzj) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorWithDeprecatedTopicOperatorTest.java]: I filter through the conditions now to find the two I'm interested in to make it more flexible.

File: design/remove-deprecated-topic-operator-from-kafka-crd.md (hunk 5 above)
Comment (scholzj) [reply design/remove-deprecated-topic-operator-from-kafka-crd.md:5]: The is not related to #3116 which is still open. This was there before the #3116 and the comments there.

**Pull request merged**